### ISSUE TITLE
Add Saline IVs

### ISSUE THREAD

**Issue by frederik-hoeft** [frederik-hoeft/rimworld-more-injuries#13]
Add craftable IVs with saline solution as an alternative to blood transfusions. There are multiple things we will need to consider here:

- there must be some disadvantages of using saline IVs over blood transfusions (you can't just replace like 50% of blood volume with salt water and expect there to be no negative health effects), this is also important for balancing
  - how would we simulate that? --> possibly by allowing saline IVs to combat hypovolemic shock, but it will result in a new hediff called [anemia](https://en.wikipedia.org/wiki/Anemia) (if that's the correct medical condition) that maybe slowly recovers over time as more blood cells are generated by the body?
  - do some more research on this to be at least sort-of scientifically accurate, and find an enjoyable way to implement this in the game
- how would saline IVs be crafted? what would be the ingredients? ~~we don't have any kind of "salt" in-game?~~ --> **use stone chunks**
- we would need new textures --> will be added in #22

**Comment by Keshash** [comment]
They could reduce bloodloss only up to a certain point. So that you can't go lower than 35% blood loss with Saline transfusions, for example.

Crafting could be from neurotamine or maybe raw meat?

Textures I can do. Already tried doing some for bloodbags (check steam dms)

**Comment by frederik-hoeft** [comment]
> They could reduce bloodloss only up to a certain point. So that you can't go lower than 35% blood loss with Saline transfusions, for example.

Yes I like that 😄  it would be able to restore low blood volume (hypovolemia / hypovolemic shock) up to a certain point. Although I'm not a fan of "hard limits" like this. We could still, technically, enforce this limit through other game mechanics like anemia (reduced number of red blood cells), where up to a certain point it will be fine to use saline IV solution but as a player you shouldn't overdo it because the anemia could become problematic at a certain point (maybe after one bag of saline, or what you said, with the 35% volume?) and after that high levels of anemia would also cause hypoxia (starvation of oxygen), just like hypovolemic shock does. At least from glancing through a few Wikipedia articles this should be somewhat accurate/believable and it would allow us to integrate more of our features with each other (hypoxia having multiple possible causes for example).
So at higher levels anemia would be just as dangerous as hypovolemic shock, at least in terms of possible organ failure, but maybe without the chance of spontaneous cardiac arrest. And if a player *does* screw up, maybe we could allow a real blood transfusion to fix the anemia as the number of red blood cells are increased by the transfusion 🤔

**Comment by frederik-hoeft** [comment]
> Crafting could be from neutroamine or maybe raw meat?

Yeah, I like neutroamine as an ingredient, although real saline is literally just salt water. So maybe actually just using stone chunks as an ingredient would work too? It's be a bit difficult to balance. We have to keep in mind that at least in terms of fighting blood loss, saline IVs should be inferior to just a normal blood transfusion, with it's benefits being potential for mass production, so it should be a cheap ingredient I think.

Although we could always lock it behind a basic research project for a better feeling of progression.

**Comment by frederik-hoeft** [comment]
> > with it's benefits being potential for mass production, so it should be a cheap ingredient I think.
> 
> The thing is, blood bags are easy to get as well. There's so many raiders in the game that can serve as walking blood bags. Perhaps, we should lock getting blood bags behind medicine technology, and allow the inferior saline bags a little earlier, with drug production?

That is a good point 🤔 Although, as it stands, aggregating a large amount of blood bags isn't that easy either, they require quite some infrastructure, like cooled storage and a large prisoner population to serve as donators. Sure, it's possible to extract blood manually from downed paws, but that's quite tedious IMO. So I wouldn't expect players in the early game being able to gather a game breaking amount of blood bags.

But you have a valid point in that we could encourage using saline IVs in the early game. We could introduce a certain risk of infection from improperly performed blood donation, and players would have to complete a research project (see #11) to investigate procedures to reduce or nullify that risk of infection. If there's a high risk of prisoners dying from infections, we could effectively discourage players from farming blood bags in early game, promoting the inferior saline IVs as an easier alternative.

**Comment by frederik-hoeft** [comment]
Reversing dependency between this and #11

**Comment by frederik-hoeft** [comment]
#12 has been merged, now waiting on #22

**Comment by frederik-hoeft** [comment]
#22 has been merged

**Comment by frederik-hoeft** [comment]
Closed by accident through GitHub actions 🙃

**Comment by frederik-hoeft** [comment]
See #52 for additional considerations
